BitwardenShared/UI/Platform/Application/AppProcessor.swift

```diff
@@ -115,22 +115,23 @@ public class AppProcessor {
     /// - Parameter url: The deep link URL to handle.
     ///
     public func openUrl(_ url: URL) async {
-        guard url.scheme?.isOtpAuthScheme == true else { return }
+        guard let scheme = url.scheme else { return }
 
-        guard let otpAuthModel = OTPAuthModel(otpAuthKey: url.absoluteString) else {
-            coordinator?.showAlert(.defaultAlert(title: Localizations.anErrorHasOccurred))
-            return
+        var route: AppRoute?
+        if scheme.isBitwardenAppScheme {
+            route = await handleBitwardenUrl(url: url)
+        } else if scheme.isOtpAuthScheme {
+            route = await handleOtpAuthUrl(url: url)
         }
+        guard let route else { return }
```

Comment:
🎨 IMO the func should be renamed given that they do not entirely handle the URL anymore but just get/create their `AppRoute`:
- `handleBitwardenUrl` -> `getBitwardenUrlRoute`
- `handleOtpAuthUrl` -> `getOtpAuthUrlRoute`

⛏️ Additionally, up to you but what would you think about moving the `scheme....` check inside the functions (returning `nil` if the scheme doesn't match) so this can be rewritten as:
```suggestion
        var route = await getBitwardenUrlRoute(url: url) ?? await getOtpAuthUrlRoute(url: url)
        guard let route else { return }
```
IMO this makes it easier to read and to scale if new url handling are needed without having a bunch of `if ... else if`

Reply:
I'm up for the name change and I'd be up for moving the scheme check as well. However, I don't think there's a way to make the cleaner nil-check work with concurrency and optimization. 

Doing this:
``` Swift
var route = await getBitwardenUrlRoute(url: url) ?? await getOtpAuthUrlRoute(url: url)
```
fails because `await` can't appear to the right of `??`. I tried moving it inside of a `guard`, but you end up with the same problem of it being an autoclosure that doesn't support concurrency.

This works, but you lose the optimization of not running subsequent calls if the first succeeds:
``` Swift
        let bitWardenRoute = await getBitwardenUrlRoute(url: url)
        let otpAuthRoute = await getOtpAuthUrlRoute(url: url)

        guard let route = bitWardenRoute ?? otpAuthRoute else { return }
```

This one gives you optimization, but I think scales the same or worse that the multiple `if else`s:
``` Swift
        var route = await getBitwardenUrlRoute(url: url)
        if route == nil {
            route = await getOtpAuthUrlRoute(url: url)
        }
        guard let route else { return }
```

Of those options, I think I lean towards the last one, or just going with the `if else` option. Thoughts?

Reply:
Oh that's a pity that it's in fact an auto closure then it can't be awaited, thanks for checking that and raising it 😞. Yep IMO the last option is fine, the code is shorter and easy to scale.

Reply:
Updated to this in the latest. 👍  Agreed that it's a shame this doesn't work. I've been bitten by this same issue a couple of times and it _feels_ like it should work and it would make the code much nicer.